modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormGuestUploadFieldUtilTest.java

```diff
@@ -115,14 +116,59 @@ public void testGuestUserAnsweringForSixthTime() throws Exception {
 		List<DDMFormInstanceRecord> ddmFormInstanceRecords = new ArrayList<>();
 
 		for (int i = 0; i < _MAXIMUM_SUBMISSIONS; i++) {
-			ddmFormInstanceRecords.add(_mockDDMFormInstanceRecord());
+			ddmFormInstanceRecords.add(_mockDDMFormInstanceRecord(_IP_ADDRESS));
 		}
 
 		_mockDDMFormInstanceLocalService(ddmFormInstanceRecords);
 
 		Assert.assertTrue(
 			DDMFormGuestUploadFieldUtil.isMaximumSubmissionLimitReached(
-				_mockDDMFormInstance(), _mockHttpServletRequest(false),
+				_mockDDMFormInstance(),
+				_mockHttpServletRequest(false, _IP_ADDRESS),
+				_MAXIMUM_SUBMISSIONS));
+	}
+
+	@Test
+	public void testGuestUsersAnsweringForDifferentTimes() throws Exception {
+		_addUploadField(true);
+
+		List<DDMFormInstanceRecord> ddmFormInstanceRecords = new ArrayList<>();
+
+		for (int i = 0; i < (_MAXIMUM_SUBMISSIONS - 1); i++) {
+			ddmFormInstanceRecords.add(
+				_mockDDMFormInstanceRecord(_IP_ADDRESS_2));
+		}
+
+		ddmFormInstanceRecords.add(_mockDDMFormInstanceRecord(_IP_ADDRESS));
+
+		_mockDDMFormInstanceLocalService(ddmFormInstanceRecords);
```

Comment:
@rijema please create a private method `_addDDMFormInstanceRecords` to reuse code since the test `testGuestUserAnsweringForFifthTime` has the same code.

Reply:
```
@Test
public void testGuestUserAnsweringForFifthTime() throws Exception {
	_addUploadField(true);

	List<DDMFormInstanceRecord> ddmFormInstanceRecords =
		_addDDMFormInstanceRecords(_IP_ADDRESS_1, _MAXIMUM_SUBMISSIONS - 1);

    .....
}

@Test
public void testGuestUserAnsweringForSixthTime() throws Exception {
	_addUploadField(true);

	List<DDMFormInstanceRecord> ddmFormInstanceRecords =
		_addDDMFormInstanceRecords(_IP_ADDRESS_1, _MAXIMUM_SUBMISSIONS);

    .....
}

@Test
public void testGuestUsersAnsweringWithDifferentIPAddresses() throws Exception {
	_addUploadField(true);

	List<DDMFormInstanceRecord> ddmFormInstanceRecords =
		_addDDMFormInstanceRecords(_IP_ADDRESS_2, _MAXIMUM_SUBMISSIONS - 1);

	ddmFormInstanceRecords.add(_mockDDMFormInstanceRecord(_IP_ADDRESS_1));

    .....
}

private List<DDMFormInstanceRecord> _addDDMFormInstanceRecords(
	String ipAddress, int size) {

	List<DDMFormInstanceRecord> ddmFormInstanceRecords = new ArrayList<>();

	for (int i = 0; i < size; i++) {
		ddmFormInstanceRecords.add(_mockDDMFormInstanceRecord(ipAddress));
	}

	return ddmFormInstanceRecords;
}
```

Reply:
I increased the private method with some other actions, please let me know if makes sense.